src/parser.c

```diff
@@ -365,12 +446,7 @@ json_object *parser_data_create_object(const char *device_id, uint8_t sensor_id,
 	 * }
 	 */
 
-	return json_msg;
-fail:
-	json_object_put(data);
-	json_object_put(json_array);
-	json_object_put(json_msg);
-	return NULL;
+	return (json_str && !has_err) ? json_str : NULL;
```

Comment:
The "else" condition should free the memory allocated for `json_string` in case of error.

Reply:
done